File: CPP/Shirt.cpp
/* Saya Alfen Fajri Nurulhaq (2201431) mengerjakan Latihan Prakitum 2 DPBO Inheritance dalam mata
   kuliah Strukdat untuk keberkahanNya maka saya tidak
   melakukan kecurangan seperti yang telah dispesikasikan.Aamiin*/
#include "Clothing.cpp"  //Clothing.cpp sebagai kelas induk (parent class)
#include <iostream>      //library IO untuk menggunakan fungsi input/output standar
#include <string>        //library string untuk menggunakan tipe data string

using namespace std;

class Shirt : public Clothing  // Tentukan kelas Shirt sebagai subkelas (derived class) dari kelas Clothing
{
private:  // Definisikan properti-properti privat
    string color;         // Properti untuk menyimpan warna kemeja
    string sleeve_type;   // Properti untuk menyimpan jenis lengan kemeja

public:
    // Method publik
    // Konstruktor
    Shirt() : Clothing()  // Konstruktor default untuk Shirt, meng-extend ke kelas Clothing
    {
        this->color = "";
        this->sleeve_type = "";
    }

    Shirt(int idProduct, string name, string brand, int price, string size, string material, string gender, string color, string sleeve_type)
        : Clothing(idProduct, name, brand, price, size, material, gender)  // Konstruktor dengan parameter, memanggil konstruktor kelas Clothing menggunakan initializer list
    {
        this->color = color;
        this->sleeve_type = sleeve_type;
    }

    // Method getter dan setter untuk color
    string getColor()
    {
        return this->color;
    }

    void setColor(string color)
    {
        this->color = color;
    }

    // Method getter dan setter untuk sleeve_type
    string getSleeve_type()
    {
        return this->sleeve_type;
    }

    void setSleeve_type(string sleeve_type)
    {
        this->sleeve_type = sleeve_type;
    }

    // Destruktor
    ~Shirt()
    {
        // Destruktor kosong dalam kasus ini
    }
};

File: PHP LATPRAK 2/Shirt.php
<?php
/* Saya Alfen Fajri Nurulhaq (2201431) mengerjakan Latihan Prakitum 2 OOP Inheritance dalam mata
   kuliah Strukdat untuk keberkahanNya maka saya tidak
   melakukan kecurangan seperti yang telah dispesikasikan.Aamiin*/
require_once 'Clothing.php';

// Mendefinisikan kelas Shirt sebagai subclass dari class Clothing dan superclass Product
class Shirt extends Clothing {
    // Mendeklarasikan variabel privat untuk color dan sleeve_type
    private $color = "";
    private $sleeve_type = "";

    // Konstruktor dengan parameter untuk inisialisasi nilai awal
    public function __construct($idProduct, $name, $brand, $price, $size, $material, $gender, $color = "", $sleeve_type = "") {
        // Memanggil konstruktor superclass (Clothing) dengan menggunakan parent::__construct()
        parent::__construct($idProduct, $name, $brand, $price, $size, $material, $gender);

        // Menginisialisasi variabel dengan nilai dari parameter konstruktor
        $this->color = $color;
        $this->sleeve_type = $sleeve_type;
    }

    // Metode untuk mendapatkan nilai color
    public function getColor() {
        return $this->color;
    }

    // Metode untuk mengatur nilai color
    public function setColor($color) {
        $this->color = $color;
    }

    // Metode untuk mendapatkan nilai sleeve_type
    public function getSleeve_type() {
        return $this->sleeve_type;
    }

    // Metode untuk mengatur nilai sleeve_type
    public function setSleeve_type($sleeve_type) {
        $this->sleeve_type = $sleeve_type;
    }

    public function __destruct()
    {
        
    }
}
?>
